abilities/music/youtube.py

```diff
@@ -81,26 +153,45 @@ def to_audio_source(song: str) -> discord.AudioSource:
             "opus",
             "pipe:1",
         ],
-        stdin=download_process.stdout,
+        stdin=binary_youtube_data_stream,
         stdout=subprocess.PIPE,
     )
     assert encoding_process.stdout
+    encoded_audio_stream = encoding_process.stdout
+
+    (
+        id,
+        title,
+        thumbnail,
+        upload_date,
+        duration,
+        view_count,
+        uploader_id,
+        uploader,
+        channel_url,
+        channel_follower_count,
+    ) = (printed_info_stream.readline().decode().strip() for _ in range(10))
+
+    return Song(
+        id=id,
+        title=title,
+        thumbnail_url=thumbnail,
+        duration=ceil(convert(duration, float, 0.0)),
+        view_count=convert(view_count, int, 0),
+        timestamp=convert(
+            upload_date,
+            lambda d: int(datetime.strptime(d, "%Y%m%d").timestamp()),
```

Comment:
Python datetime objects can be naive or timezone-aware. While an aware object represents a specific moment in time, a naive object does not contain enough information to unambiguously locate itself relative to other datetime objects. Since this can lead to errors, it is recommended to always use timezone-aware objects.
```suggestion
            lambda d: int(
                datetime.strptime(d, "%Y%m%d").replace(tzinfo=timezone.utc).timestamp(),
            ),
```